### DISCUSSION TITLE
Timescale DB Support

### DISCUSSION THREAD

**Discussion by all2pie** [mikro-orm/mikro-orm#2226] [Ideas]
Please add support for [TImesclae DB](https://docs.timescale.com/timescaledb/latest/quick-start/node/#prerequisites), if it is already supported then please add it in Docs.

**Comment by tukusejssirs** [comment]
TimescaleDB is an extension of PostgreSQL, thus the SQL stuff is already supported by MikroORM.

All we need to add support for is the [TimescaleDB extension API](https://docs.timescale.com/api/latest/?utm_source=timescaledb&utm_medium=github&utm_campaign=community#apis-grouped-by-related-feature), which can already be used using raw queries in MikroORM. That said, it would be nice to add direct support of those APIs in MikroORM.

@B4nan, what would be your idea? What the TimescaleDB in MikroORM should look like? We could add a separate ‘driver’ for it, however, it might be too much due to the fact that it is only an extension of PostgreSQL. Could we simply add support for [PostgreSQL extensions](https://www.postgresql.org/download/products/6-postgresql-extensions/)? This discussion deals with the TimescaleDB extension, someone else might want to add support for [PostGis](https://www.postgis.net/) or any other extension.

WDYT?

**Reply by tukusejssirs** [reply]
@B4nan, here are some related Prisma issues:

- prisma/prisma#3228;
- prisma/prisma#12308.
